turtle-1.5.15 no longer compiles with ghc-8.2.2 or older

I'm getting errors like the following:

~~~~
src/Turtle/Bytes.hs:790:32: error: 
    • Variable not in scope: (<>) :: a -> p -> t1 
    • Perhaps you meant one of these: 
        ‘<$>’ (imported from Prelude), ‘*>’ (imported from Prelude), 
        ‘<$’ (imported from Prelude) 
    | 
790 |             let bytes = prefix <> suffix 
    |                                ^^ 
~~~~

See https://travis-ci.org/peti/distribution-opensuse/builds/589054086 for more details.

I suppose this is fine, but there should be a  version constraint on `base` to prevent cabal from picking that version of `turtle` when compiling with ghc 8.2.2 or earlier.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,2 +1,2 @@
 language: nix
-script: nix-build release.nix
+script: NIX_PATH= nix-build release.nix

--- a/default.nix
+++ b/default.nix
@@ -6,22 +6,23 @@ let
 
     sha256 = "1j52yvkhw1inp6ilpqy81xv1bbwgwqjn0v9647whampkqgn6dxhk";
   };
+
   readDirectory = import ./nix/readDirectory.nix;
-
-  overrides =
-    let
-      manualOverrides = haskellPackagesNew: haskellPackagesOld: {
-      };
-
-    in
-      pkgs.lib.composeExtensions (readDirectory ./nix) manualOverrides;
 
   config = {
     packageOverrides = pkgs: {
       haskell = pkgs.haskell // {
         packages = pkgs.haskell.packages // {
           "${compiler}" = pkgs.haskell.packages."${compiler}".override {
-            inherit overrides;
+            overrides =
+              let
+                manualOverrides = haskellPackagesNew: haskellPackagesOld: {
+                  system-fileio =
+                    pkgs.haskell.lib.dontCheck haskellPackagesOld.system-fileio;
+                };
+
+              in
+                pkgs.lib.composeExtensions (readDirectory ./nix) manualOverrides;
           };
         };
       };

--- a/default.nix
+++ b/default.nix
@@ -1,3 +1,5 @@
+{ compiler ? "ghc864" }:
+
 let
   nixpkgs = builtins.fetchTarball {
     url = "https://github.com/NixOs/nixpkgs/archive/312a059bef8b29b4db4e73dc02ff441cab7bb26d.tar.gz";
@@ -6,16 +8,22 @@ let
   };
   readDirectory = import ./nix/readDirectory.nix;
 
+  overrides =
+    let
+      manualOverrides = haskellPackagesNew: haskellPackagesOld: {
+      };
+
+    in
+      pkgs.lib.composeExtensions (readDirectory ./nix) manualOverrides;
+
   config = {
     packageOverrides = pkgs: {
-      haskellPackages = pkgs.haskellPackages.override {
-        overrides =
-          let
-            manualOverrides = haskellPackagesNew: haskellPackagesOld: {
-            };
-
-          in
-            pkgs.lib.composeExtensions (readDirectory ./nix) manualOverrides;
+      haskell = pkgs.haskell // {
+        packages = pkgs.haskell.packages // {
+          "${compiler}" = pkgs.haskell.packages."${compiler}".override {
+            inherit overrides;
+          };
+        };
       };
     };
   };
@@ -24,7 +32,9 @@ let
     import nixpkgs { inherit config; };
 
 in
-  { inherit (pkgs.haskellPackages) turtle;
+  { inherit (pkgs.haskell.packages."${compiler}") turtle;
 
-    shell = (pkgs.haskell.lib.doBenchmark pkgs.haskellPackages.turtle).env;
+    shell = (pkgs.haskell.packages."${compiler}".turtle).env;
+
+    inherit (pkgs.releaseTools) aggregate;
   }

--- a/release.nix
+++ b/release.nix
@@ -1,5 +1,17 @@
 let
-  default = import ./default.nix;
+  ghc822 = import ./default.nix { compiler = "ghc822"; };
+
+  default = import ./default.nix { };
 
 in
-  { inherit (default) turtle; }
+  { turtle =
+      default.aggregate
+        { name = "turtle";
+
+          constituents = [
+            ghc822.turtle
+
+            default.turtle
+          ];
+        };
+  }

--- a/release.nix
+++ b/release.nix
@@ -6,7 +6,7 @@ let
 in
   { turtle =
       default.aggregate
-        { name = "turtle";
+        { name = "all";
 
           constituents = [
             ghc822.turtle

--- a/src/Turtle/Bytes.hs
+++ b/src/Turtle/Bytes.hs
@@ -49,6 +49,7 @@ import Control.Concurrent.Async (Async, Concurrently(..))
 import Control.Monad.IO.Class (MonadIO(..))
 import Control.Monad.Managed (MonadManaged(..))
 import Data.ByteString (ByteString)
+import Data.Monoid
 import Data.Streaming.Zlib (Inflate, Popper, PopperRes(..), WindowBits(..))
 import Data.Text (Text)
 import Data.Text.Encoding (Decoding(..))

